admin/app/components/solidus_admin/users/addresses/component.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class SolidusAdmin::Users::Addresses::Component < SolidusAdmin::BaseComponent
+  include SolidusAdmin::Layout::PageHelpers
+
+  def initialize(user:, address: nil, type: nil)
+    @user = user
+    @bill_address = bill_address(address, type)
+    @ship_address = ship_address(address, type)
+  end
+
+  def form_id
+    @form_id ||= "#{stimulus_id}--form-#{@user.id}"
+  end
+
+  def tabs
+    [
+      {
+        text: t('.account'),
+        href: solidus_admin.user_path(@user),
+        current: false,
+      },
+      {
+        text: t('.addresses'),
+        href: solidus_admin.addresses_user_path(@user),
+        current: true,
+      },
+      {
+        text: t('.order_history'),
+        href: solidus_admin.orders_user_path(@user),
+        current: false,
+      },
+      {
+        text: t('.items'),
+        href: spree.items_admin_user_path(@user),
+        current: false,
+      },
+      {
+        text: t('.store_credit'),
+        href: spree.admin_user_store_credits_path(@user),
+        current: false,
+      },
+    ]
+  end
+
+  def last_login(user)
+    return t('.last_login.never') if user.try(:last_sign_in_at).blank?
+
+    t(
+      '.last_login.login_time_ago',
+      # @note The second `.try` is here for the specs and for setups that use a
+      # custom User class which may not have this attribute.
+      last_login_time: time_ago_in_words(user.try(:last_sign_in_at))
+    ).capitalize
+  end
```

Comment:
Would it make sense to extract this into a `module` we can `include` in all user tabs components? It seems a bit repetitive to have to implement this in all user related components

Reply:
Yeah definitely, I can extract that as part of the items page if that's alright, working on that one tomorrow.